app/models/image.rb

```diff
@@ -972,26 +972,81 @@ def self.update_copyright_holder(old_name, new_name, user)
     # This is orders of magnitude faster than doing via active-record.
     old_name = Image.connection.quote(old_name)
     new_name = Image.connection.quote(new_name)
-    data = Image.connection.select_rows(%(
-      SELECT id, YEAR(`when`), license_id FROM images
-      WHERE user_id = #{user.id} AND copyright_holder = #{old_name}
-    ))
+
+    select_manager = arel_select_copyright_holder(old_name, user)
+    # puts(select_manager.to_sql)
+    data = Image.connection.select_rows(select_manager.to_sql)
     return unless data.any?
 
     # brakeman generates what appears to be a false positive SQL injection
     # warning.  See https://github.com/presidentbeef/brakeman/issues/1231
-    Image.connection.insert(%(
-      INSERT INTO copyright_changes
-        (user_id, updated_at, target_type, target_id, year, name, license_id)
-      VALUES
-        #{data.map do |id, year, lic|
-            "(#{user.id},NOW(),'Image',#{id},#{year},#{old_name},#{lic})"
-          end.join(",\n")}
-    ))
-    Image.connection.update(%(
-      UPDATE images SET copyright_holder = #{new_name}
-      WHERE user_id = #{user.id} AND copyright_holder = #{old_name}
-    ))
+    insert_manager = arel_insert_copyright_changes(data, old_name, user)
+    # puts(insert_manager.to_sql)
+    Image.connection.insert(insert_manager.to_sql)
+
+    update_manager = arel_update_copyright(old_name, new_name, user)
+    # puts(update_manager.to_sql)
+    Image.connection.update(update_manager.to_sql)
+  end
+
+  # SELECT id, YEAR(`when`), license_id FROM images
+  # WHERE user_id = #{user.id} AND copyright_holder = #{old_name}
+  private_class_method def self.arel_select_copyright_holder(old_name, user)
+    i = Image.arel_table
+    i.project([i[:id], i[:when].year, i[:license_id]]).
+      where(i[:user_id].eq(user.id).
+            and(i[:copyright_holder].eq(old_name)))
+  end
+
+  # INSERT INTO copyright_changes
+  #   (user_id, updated_at, target_type, target_id, year, name, license_id)
+  # VALUES
+  #   #{data.map do |id, year, lic|
+  #       "(#{user.id},NOW(),'Image',#{id},#{year},#{old_name},#{lic})"
+  #     end.join(",\n")}
+  private_class_method def self.arel_insert_copyright_changes(
+    data, old_name, user
+  )
+    cc = CopyrightChange.arel_table
+    values_list = arel_values_list_copyright_changes(data, old_name, user)
+    Arel::InsertManager.new.tap do |manager|
+      manager.into(cc)
+      manager.columns << cc[:user_id]
+      manager.columns << cc[:updated_at]
+      manager.columns << cc[:target_type]
+      manager.columns << cc[:target_id]
+      manager.columns << cc[:year]
+      manager.columns << cc[:name]
+      manager.columns << cc[:license_id]
+      manager.values = manager.create_values_list(values_list)
+    end
+  end
```

Comment:
Yikes, what a mouthful.  I would love to see if Joe can find an AR equivalent that will bulk-add records to a table.  Unfortunately, adding these one at a time is not an option.  That's what the "orders of magnitude faster" comment in the original code refers to.

Reply:
Not out of the box, at least not until Rails 6.  See this thread:

  https://stackoverflow.com/questions/10022178/insert-multiple-records-using-ruby-on-rails-active-record